asterisk_click2dial/asterisk_click2dial.py

```diff
@@ -429,15 +397,19 @@ class phone_common(orm.AbstractModel):
 
     def click2dial(self, cr, uid, erp_number, context=None):
         if not erp_number:
-            orm.except_orm(
+            raise orm.except_orm(
                 _('Error:'),
                 _('Missing phone number'))
 
         user, ast_server, ast_manager = \
             self.pool['asterisk.server']._connect_to_asterisk(
                 cr, uid, context=context)
-        ast_number = self.pool['asterisk.server']._reformat_number(
-            cr, uid, erp_number, ast_server, context=context)
+        ast_number = self.convert_to_dial_number(erp_number)
```

Comment:
self is not a record set

Either you need to convert this function to api8 or you can add cr, uid, context